core/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -505,16 +492,17 @@ public void deleteClosedIndex(String reason, IndexMetaData metaData, ClusterStat
     public void deleteIndexStore(String reason, IndexMetaData metaData, ClusterState clusterState, boolean closed) throws IOException {
         if (nodeEnv.hasNodeFile()) {
             synchronized (this) {
-                String indexName = metaData.getIndex().getName();
-                if (indices.containsKey(indexName)) {
-                    String localUUid = indices.get(indexName).indexUUID();
-                    throw new IllegalStateException("Can't delete index store for [" + indexName + "] - it's still part of the indices service [" + localUUid + "] [" + metaData.getIndexUUID() + "]");
+                Index index = metaData.getIndex();
+                if (hasIndex(index)) {
+                    String localUUid = indexService(index).indexUUID();
+                    throw new IllegalStateException("Can't delete index store for [" + index.getName() + "] - it's still part of the indices service [" + localUUid + "] [" + metaData.getIndexUUID() + "]");
```

Comment:
we can just add `index` to the message and be done. the only situation we have where the uuids are not the same is when `indexService(index).indexUUID() != index.getUUID()` in which case things are seriously broken. just cleans up the code a bit - no big deal - was just sharing happiness ;)